torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -130,7 +131,16 @@ IValue UserRRef::toHere(const float timeoutSeconds) const {
       "RRef creation via rpc.remote() timed out, and it "
       "is possible that the RRef on the owner node does not exist.");
   // see Note [Best-Effort Check on Deleted UserRRefs]
-  RECORD_USER_SCOPE("to_here");
+  auto toHereKey = fmt::format(
+      "to_here#({})->({})",
+      RpcAgent::getCurrentRpcAgent()->getWorkerInfo().name_,
+      RpcAgent::getCurrentRpcAgent()->getWorkerInfo(ownerId_).name_);
+  if (torch::autograd::profiler::profilerEnabled()) {
+    auto& remoteProfilerManager =
+        torch::distributed::rpc::RemoteProfilerManager::getInstance();
+    remoteProfilerManager.setCurrentKey(toHereKey);
+  }
+  RECORD_USER_SCOPE(toHereKey);
```

Comment:
Curious, why `toHereKey` and `RECORD_USER_SCOPE` needs to be out of the if clause above? Do we need `RECORD_USER_SCOPE` when the profiler is not enabled?

Reply:
We don't need `RECORD_USER_SCOPE` when the profiler isn't enabled, although, when it is not enabled, there isn't significant overhead and it is a no-op.

The following won't work: 
```
if (profilerEnabled()) {
    RECORD_USER_SCOPE(...)
}
```
because the guard will only apply for the scope it is in, so it will be destructed at the end of the if block, but we want it to persist across the entire function.

Although I agree, since constructing `toHereKey` is somewhat expensive (involves a few gets and string concatenation), we should only have this if the profiler is enabled.